graphql_compiler/schema_generation/sqlalchemy/sqlalchemy_reflector.py

```diff
@@ -0,0 +1,142 @@
+import re
+
+from sqlalchemy import Column, Table, text
+from sqlalchemy.dialects.mssql.base import ischema_names
+
+
+def fast_sql_server_reflect(engine, metadata, schema, primary_key_selector=None):
+    """Reflect the metadata of a SQL Server in a quick but shallow manner.
+
+    This function is roughly a faster, but shallower equivalent to the SQLAlchemy metadata.reflect()
+    method. It currently only reflects: columns, column types and primary keys. Other information,
+    like foreign key metadata, is currently ignored but might be inferred in the future.
+
+    Args:
+        engine: SQLAlchemy Engine, engine connected to a SQL server. It must connected to the master
+                database in the SQL Server.
+        metadata: MetaData object to reflect the metadata of the SQL Server to.
+        schema: string in the format <databaseName>.<schemaName> specifying the schema
+                to reflect into the metadata.
+        primary_key_selector: optional function that takes in a table name and list of dicts
+                              specifying column metadata and returns a set of column names to use
+                              as the primary key for the corresponding SQLAlchemy Table object.
+                              The compiler requires each SQLAlchemy Table to have a primary key.
+                              The primary keys do not need to be the primary keys in the underlying
+                              tables. They just need to be an unique and non-null identifier of
+                              each row. This parameter should be used to amend SQLAlchemy Table
+                              objects with such non-explicitly enforced primary keys.
+    """
```

Comment:
Can you return a list of ignored types, a list of primary keys that were inferred, etc. to make the process more transparent? If you run a production system and realize days after generating the schema that it was incorrect, you can't reproduce the old state to find what went wrong. You can only rely on logs and dumps.